textpipe/doc.py

```diff
@@ -375,3 +375,35 @@ def keyterms(self):
         [('awesome', 0.32456160227748454), ('capital', 0.32456160227748454), ('amsterdam', 0.17543839772251532)]
         """
         return self.extract_keyterms()
+
+    @property
+    def vectors(self):
+        """
+        Returns word embeddings for the words in the document.
+        The used spacy models don't have "true" word vectors
+        but only context-sensitive tensors that are within the document.
+
+        Returns:
+        A dictionary mapping words from the document to a 4-tuple with the
+        corresponding values of the following variables:
+
+        has vector: Does the token have a vector representation?
+        vector norm: The L2 norm of the token's vector (the square root of the
+                    sum of the values squared)
+        OOV: Out-of-vocabulary (This variable always gets the vlue TRUE since
+                                there are no vectors included in the model)
+        vector: The vector represenation of the word
+
+        >>> doc = Doc('Test sentence')
+        >>> doc.vectors['Test'][2]
+        True
+        >>> doc.vectors['Test'][3].shape[-1]
+        384
+        >>> doc.vectors['Test'][1] == doc.vectors['sentence'][1]
+        False
+        """
+        return {token.text: (token.has_vector,
+                             token.vector_norm,
+                             token.is_oov,
+                             token.vector)
+                for token in self._spacy_doc}
```

Comment:
If we are still using the `self._spacy_doc` object here, it will still use the spaCy's small models currently (e.g. `en_core_web_sm` and `nl_core_news_sm` as indicated).

What we did instead in our SJTextpipe is that we applied a new spaCy nlp model to the `nlp(doc.clean)` for the word embeddings operation. That will give us the real vectors. External word embedding models are based on [fastText ](https://github.com/facebookresearch/fastText/blob/master/docs/crawl-vectors.md). Check the instruction here: https://spacy.io/usage/vectors-similarity#converting

Reply:
Sounds pretty solid. Could you share your code for this with at least @KleinRana? Then we can come up with a plan to consolidate this.

Reply:
I will change self._spacy_doc to self._load_spacy_doc(lang, model_name) such that external models can be used